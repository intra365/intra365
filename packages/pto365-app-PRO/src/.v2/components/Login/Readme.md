
## Authenticated 


```js
<Login isAuthenticated />
```

## Not Authenticated 


```js
<Login isAuthenticated={false} />
```

```js
<Authorized tenant={mocks.tenant()} />
```
