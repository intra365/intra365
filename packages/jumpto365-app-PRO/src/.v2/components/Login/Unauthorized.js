import React, { Component } from 'react';
import Util from "../../../util"
import { CompoundButton,DefaultButton, IButtonProps } from 'office-ui-fabric-react/lib/Button';
import { ProgressIndicator } from 'office-ui-fabric-react/lib/ProgressIndicator';

import { isAuthenticated,userName,userId } from '../../../components/Auth'

import ProcessPage from '../../../components/Process/ProcessPage'
import  SuggestTenantOwner from '../../../components/Process/SuggestTenantOwner'
import  SetupTenant from '../../../components/Process/SetupTenant'


export default class Unauthorized extends Component {
  constructor(props) {
    super(props);
    this.state = {
        tenant : props.tenant,
        prompts: {body : "",// "Your are the first to visit JumpTo365 from your Office 365 tenant, so you have been designated as owner. ",
               action1Title : "",// "Configure",
               action1 : "",// "We have some questions for you which impacts the experience.",
               action2Title : "",// "If you know that another one within your company should be the owner",
               action2 : "",// "Change Owner",
               buttonLogin : "Login" 

      } 
    };

  }

  componentDidMount = () => {
       Util.Prompts.get("login/unauthorized", {})
       .then(prompts => {
        this.setState({
          loaded:true,
          prompts
        })
      })
      .catch((error) => {
        this.setState({
          error
        })
      })
  }
  
  render() {
    if (this.state.error){
        return <div>{JSON.stringify(this.state.error)}</div>
    }
    if (!this.state.loaded){
        return <ProgressIndicator />
    }

    return (
      <ProcessPage title="Login" options={<DefaultButton
        primary={true}
        data-automation-id='test'
        checked={false}
        text={this.state.prompts.buttonLogin}
        onClick={this.props.onLogin}
        style={{ minWidth: '120px',margin:"10px" }}
      />}>
      <div dangerouslySetInnerHTML={{ __html: this.state.prompts.html }}></div>
        </ProcessPage>
        
        
    )
   
  }
}


