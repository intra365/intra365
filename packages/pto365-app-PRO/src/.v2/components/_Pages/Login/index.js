import React, { Component } from 'react';
import PropTypes from 'prop-types'
import PageLayoutMain from '../../_Layouts/PageLayoutMain';
import PageHeader from '../../PageHeader';
import Login  from '../../Login';
import Jumpto365App from "../../_Contexts/Jumpto365App"
import PageBody from '../../PageBody';

/**
 * Describe overall purpose of the component
 *
 * @export
 * @class LoginPage
 * @extends {Component}
 */
export default class LoginPage extends Component {

    static propTypes  = {
        context : PropTypes.object   // sample - remove or rename
    }
    

    constructor(props) {
        super(props);
        // if (this.props.location && this.props.location.hash && this.props.context){
        //         Jumpto365App.registerAuthToken(this.props.context,this.props.location.hash)

        // }
    }


    render() {
        if (this.props.context == null){
            return null
        }
        return (
            
            <PageLayoutMain>
                <PageHeader title="Login" color="#2a7ab9"/>
                <PageBody>
                <Login context={this.props.context}/>
                </PageBody>
            </PageLayoutMain>
       
        )
    }
}

