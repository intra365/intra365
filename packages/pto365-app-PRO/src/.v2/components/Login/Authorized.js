import React, { Component } from 'react';
import Util from "../../../util"
import { CompoundButton,DefaultButton, IButtonProps } from 'office-ui-fabric-react/lib/Button';
import { ProgressIndicator } from 'office-ui-fabric-react/lib/ProgressIndicator';
import { navigate ,Location} from "@reach/router"
import { isAuthenticated,userName,userId } from '../../../components/Auth'
import PropTypes from 'prop-types'

import ProcessPage from '../../../components/Process/ProcessPage'
import  SuggestTenantOwner from '../../../components/Process/SuggestTenantOwner'
import  SetupTenant from '../../../components/Process/SetupTenant'
import { jumpto365API } from '../../../api/pto365'

export default class Authorized extends Component {

  
  static propTypes  = {
    tenant : PropTypes.object.isRequired,
    upn : PropTypes.object.isRequired,
    onLogout : PropTypes.func
}


  constructor(props) {
    super(props);
    this.state = {prompts: {},tenant:{stage:"NEW"}}

  }
  componentWillMount = () => {
    navigate(`/dashboard`)
    //navigate(`/@/${this.props.upn}`)
  }
  componentDidMount = () => {

    Util.Prompts.get("login/authorized" , this.props.tenant)
    .then(prompts => {
      this.setState({
        loaded:true,
        prompts
      })
    })
    .catch((error) => {
      this.setState({
        error: error.message
      })
    })
  }
  
  render() {
    if (this.state.error){
        return <div>{JSON.stringify(this.state.error)}</div>
    }
    if (!this.state.loaded){
        return <ProgressIndicator />
    }


    switch (this.state.tenant.stage.toUpperCase()) {
        case "NEW":
                var options = []
                
                options.push(<SetupTenant tenant={this.state.tenant.name} title={this.state.prompts.action1title} description={this.state.prompts.action1} />)
                options.push(<SuggestTenantOwner tenant={this.state.tenant.name} title={this.state.prompts.action2title} description={this.state.prompts.action2} />)

                return <ProcessPage title={this.state.prompts.title} options={options}>
                
                <div dangerouslySetInnerHTML={{ __html: this.state.prompts.html }}></div>
                </ProcessPage>
            
        break;
              
        default:
            return null
            break;
    }
  }
}

