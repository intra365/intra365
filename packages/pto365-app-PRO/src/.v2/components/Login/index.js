import React, { Component } from 'react';
import { UserAgentApplication } from 'msal';
import PropTypes from 'prop-types'

import { isAuthenticated } from '../../../components/Auth'
import  applicationConfig  from "../../../components/Auth/config.json"
import {pto365Authenticate,pto365Track} from '../../../api/pto365'
import Authorized from './Authorized'
import Unauthorized from './Unauthorized'
import DialoguePageTemplate  from '../../../pages/_Templates/DialoguePageTemplate'
import { getUserAgentApplcation } from "../../utilities/Auth"
import Jumpto365App from '../_Contexts/Jumpto365App'
import { navigate } from '../../../../node_modules/@reach/router/lib/history';

export default class Login extends Component {


  static propTypes  = {
    context : PropTypes.object,
    onLogin : PropTypes.func
   
}
componentDidMount = () => {
//   if (this.props.context && this.props.context.isAuthenticated && this.props.context.userId){
//   navigate(`/@/${this.props.context.userId}`)
// }
}

  render() {

    if (!this.props.context) return "Missing context"
    
    if (this.props.context.isAuthenticated){
      return (<DialoguePageTemplate><Authorized tenant={this.props.context.tenant} upn={this.props.context.userId} /></DialoguePageTemplate>)
    }else{
      return (<DialoguePageTemplate><Unauthorized onLogin={()=>{ Jumpto365App.login(this.props.context)}} /></DialoguePageTemplate>)
    }

  }
}

