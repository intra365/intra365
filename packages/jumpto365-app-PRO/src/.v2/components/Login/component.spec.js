import React, { Component } from 'react';

import Login from '.'
import Authorized from './Authorized'

test('Check that login renders', () => {

    const wrapper = shallow(<Login  isAuthenticated={false} tenant={mocks.tenant()}/>);

    expect(wrapper).toMatchSnapshot();
 
}

)

test('Check that it renders', () => {

    const wrapper = shallow(<Authorized tenant={mocks.tenant()}/>);

    expect(wrapper).toMatchSnapshot();
 
}

)