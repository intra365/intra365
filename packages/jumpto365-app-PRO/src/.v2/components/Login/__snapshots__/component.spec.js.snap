// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Authorized
    tenant={
        Object {
            "name": "Jumpto365",
            "properties": Object {
              "logo": "https://placeimg.com/100/20/any",
            },
            "stage": "new",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Check that login renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    isAuthenticated={false}
    tenant={
        Object {
            "name": "Jumpto365",
            "properties": Object {
              "logo": "https://placeimg.com/100/20/any",
            },
            "stage": "new",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "Missing context",
  Symbol(enzyme.__nodes__): Array [
    "Missing context",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
